event_engine build failure: cfstream_endpoint.cc: error: 'CFErrorDomain' was not declared in this scope; did you mean 'CFErrorGetDomain'?

```
[ 11%] Building CXX object CMakeFiles/grpc_authorization_provider.dir/src/core/lib/event_engine/default_event_engine_factory.cc.o
/opt/local/bin/g++-mp-12 -DGPR_BACKWARDS_COMPATIBILITY_MODE -DPROTOBUF_USE_DLLS -Dgrpc_authorization_provider_EXPORTS -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/include -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/address_sorting/include -I/opt/local/libexec/openssl3/include -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/ext/upb-generated -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/ext/upbdefs-generated -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/upb -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/utf8_range -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/xxhash -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/build/gens -isystem /opt/local/include -pipe -I/opt/local/libexec/openssl3/include -Os -I/opt/local/libexec/protobuf3-cpp-upstream/include -DNDEBUG -I/opt/local/libexec/openssl3/include -isystem/opt/local/include/LegacySupport -isystem/opt/local/include -D__STDC_FORMAT_MACROS -D_GLIBCXX_USE_CXX11_ABI=0  -std=gnu++14 -arch ppc -mmacosx-version-min=10.6 -fPIC -MD -MT CMakeFiles/grpc_authorization_provider.dir/src/core/lib/event_engine/default_event_engine_factory.cc.o -MF CMakeFiles/grpc_authorization_provider.dir/src/core/lib/event_engine/default_event_engine_factory.cc.o.d -o CMakeFiles/grpc_authorization_provider.dir/src/core/lib/event_engine/default_event_engine_factory.cc.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/default_event_engine_factory.cc
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc: In function 'absl::lts_20230125::Status grpc_event_engine::experimental::{anonymous}::CFErrorToStatus(grpc_event_engine::experimental::CFTypeUniqueRef<__CFError*>)':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc:34:3: error: 'CFErrorDomain' was not declared in this scope; did you mean 'CFErrorGetDomain'?
   34 |   CFErrorDomain cf_domain = CFErrorGetDomain((cf_error));
      |   ^~~~~~~~~~~~~
      |   CFErrorGetDomain
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc:39:22: error: 'cf_domain' was not declared in this scope
   39 |   CFStringGetCString(cf_domain, domain_buf, 256, kCFStringEncodingUTF8);
      |                      ^~~~~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc: In member function 'void grpc_event_engine::experimental::CFStreamEndpointImpl::Connect(absl::lts_20230125::AnyInvocable<void(absl::lts_20230125::Status)>, grpc_event_engine::experimental::EventEngine::ResolvedAddress)':
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc:117:58: error: 'QOS_CLASS_DEFAULT' was not declared in this scope
  117 |                                dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0));
      |                                                          ^~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc:117:32: error: 'dispatch_get_global_queue' was not declared in this scope
  117 |                                dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0));
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc:116:3: error: 'CFReadStreamSetDispatchQueue' was not declared in this scope
  116 |   CFReadStreamSetDispatchQueue(cf_read_stream_,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 11%] Building CXX object CMakeFiles/grpc.dir/src/core/ext/filters/client_channel/client_channel_factory.cc.o
/opt/local/bin/g++-mp-12 -DGPR_BACKWARDS_COMPATIBILITY_MODE -Dgrpc_EXPORTS -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/include -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0 -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/address_sorting/include -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/ext/upb-generated -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/ext/upbdefs-generated -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/upb -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/utf8_range -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/third_party/xxhash -isystem /opt/local/libexec/openssl3/include -isystem /opt/local/include -pipe -I/opt/local/libexec/openssl3/include -Os -I/opt/local/libexec/protobuf3-cpp-upstream/include -DNDEBUG -I/opt/local/libexec/openssl3/include -isystem/opt/local/include/LegacySupport -isystem/opt/local/include -D__STDC_FORMAT_MACROS -D_GLIBCXX_USE_CXX11_ABI=0  -std=gnu++14 -arch ppc -mmacosx-version-min=10.6 -fPIC -MD -MT CMakeFiles/grpc.dir/src/core/ext/filters/client_channel/client_channel_factory.cc.o -MF CMakeFiles/grpc.dir/src/core/ext/filters/client_channel/client_channel_factory.cc.o.d -o CMakeFiles/grpc.dir/src/core/ext/filters/client_channel/client_channel_factory.cc.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/ext/filters/client_channel/client_channel_factory.cc
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_grpc/grpc/work/grpc-1.57.0/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc:118:3: error: 'CFWriteStreamSetDispatchQueue' was not declared in this scope
  118 |   CFWriteStreamSetDispatchQueue(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/grpc_authorization_provider.dir/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```
What is going wrong here?

P. S. Unfortunately, I may some somewhat late with reporting the bug, since the last version I built was 1.54.2, so I am not sure when exactly this has been introduced.

--- a/src/core/lib/event_engine/cf_engine/cf_engine.cc
+++ b/src/core/lib/event_engine/cf_engine/cf_engine.cc
@@ -15,6 +15,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include <CoreFoundation/CoreFoundation.h>
 
@@ -215,4 +217,5 @@ EventEngine::TaskHandle CFEventEngine::RunAfterInternal(
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE

--- a/src/core/lib/event_engine/cf_engine/cf_engine.h
+++ b/src/core/lib/event_engine/cf_engine/cf_engine.h
@@ -16,6 +16,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include <grpc/event_engine/event_engine.h>
 
@@ -81,6 +83,7 @@ class CFEventEngine : public EventEngine,
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE
 
 #endif  // GRPC_SRC_CORE_LIB_EVENT_ENGINE_CF_ENGINE_CF_ENGINE_H

--- a/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc
+++ b/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc
@@ -15,6 +15,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include "src/core/lib/event_engine/cf_engine/cfstream_endpoint.h"
 #include "src/core/lib/event_engine/trace.h"
@@ -351,4 +353,5 @@ void CFStreamEndpointImpl::DoWrite(
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE

--- a/src/core/lib/event_engine/cf_engine/cfstream_endpoint.h
+++ b/src/core/lib/event_engine/cf_engine/cfstream_endpoint.h
@@ -16,6 +16,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include <CoreFoundation/CoreFoundation.h>
 
@@ -141,6 +143,7 @@ class CFStreamEndpoint : public EventEngine::Endpoint {
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE
 
 #endif  // GRPC_SRC_CORE_LIB_EVENT_ENGINE_CF_ENGINE_CFSTREAM_ENDPOINT_H

--- a/src/core/lib/event_engine/cf_engine/cftype_unique_ref.h
+++ b/src/core/lib/event_engine/cf_engine/cftype_unique_ref.h
@@ -16,6 +16,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include <CoreFoundation/CoreFoundation.h>
 
@@ -74,6 +76,7 @@ class CFTypeUniqueRef {
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE
 
 #endif  // GRPC_SRC_CORE_LIB_EVENT_ENGINE_CF_ENGINE_CFTYPE_UNIQUE_REF_H

--- a/src/core/lib/event_engine/cf_engine/dns_service_resolver.cc
+++ b/src/core/lib/event_engine/cf_engine/dns_service_resolver.cc
@@ -15,6 +15,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include "absl/strings/str_format.h"
 
@@ -226,4 +228,5 @@ void DNSServiceResolverImpl::Shutdown() {
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE

--- a/src/core/lib/event_engine/cf_engine/dns_service_resolver.h
+++ b/src/core/lib/event_engine/cf_engine/dns_service_resolver.h
@@ -16,6 +16,8 @@
 #include <grpc/support/port_platform.h>
 
 #ifdef GPR_APPLE
+#include <AvailabilityMacros.h>
+#ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
 #include <CoreFoundation/CoreFoundation.h>
 #include <dns_sd.h>
@@ -112,6 +114,7 @@ class DNSServiceResolver : public EventEngine::DNSResolver {
 }  // namespace experimental
 }  // namespace grpc_event_engine
 
+#endif  // AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 #endif  // GPR_APPLE
 
 #endif  // GRPC_SRC_CORE_LIB_EVENT_ENGINE_CF_ENGINE_DNS_SERVICE_RESOLVER_H

